### ISSUE TITLE
fastq-dump.2.9.0 sys: timeout exhausted while reading file within network system module

### ISSUE THREAD

**Issue by s-tahara** [pitagora-network/pitagora-cwl#38]
from #36 

Environment  : NIG supercomputer

I wanna get fastq file as follows:
```
cwltool --singularity --debug "https://raw.githubusercontent.com/pitagora-network/pitagora-cwl/master/workflows/download-fastq/download-fastq.cwl" "ERX2868692_job.yml"
```
ERX2868692_job.yml :
```
run_ids:
- ERR2862454
nthreads: 4
repo: ddbj
```
However, returned `fastq-dump` network error

```
2021-03-19T10:08:00 fastq-dump.2.9.0 sys: timeout exhausted while reading file within network system module - mbedtls_ssl_read returned -76 ( NET - Reading information from the socket failed )
2021-03-19T10:14:06 fastq-dump.2.9.0 sys: timeout exhausted while reading file within network system module - mbedtls_ssl_read returned -76 ( NET - Reading information from the socket failed )
2021-03-19T10:14:52 fastq-dump.2.9.0 sys: timeout exhausted while creating file within network system module - Failed to Make Connection in KClientHttpOpen to 'sra-download.ncbi.nlm.nih.gov:443
2021-03-19T10:14:52 fastq-dump.2.9.0 err: timeout exhausted while creating file within network system module - error with https open 'https://sra-download.ncbi.nlm.nih.gov/traces/refseq/NC_000004.12'
2021-03-19T10:17:40 fastq-dump.2.9.0 sys: timeout exhausted while reading file within network system module - mbedtls_ssl_read returned -76 ( NET - Reading information from the socket failed )
```

I searched a little about this error and found someone recommended new version (fastq-dump 2.10.9) (https://github.com/ncbi/sra-tools/issues/228)
but I don't know what I should do the next.
If you know about this error or another solution, please let me know.

Thank you for your kindness.

**Comment by inutano** [comment]
Sorry for closing this issue by mistake. I've been investigating for a while about this problem and found that this is a kind of problem which won't be solved straightforwardly.

First, your suggestion is correct. The version of sra-tools causes the format conversion failure. So it should be solved by upgrading the tool version of this workflow.

But here's another problem: the newer version of sra-tools has an issue when running in a workflow platform like CWL. It requires an "interactive" configuration to set the behavior of the tool. I've found a solution to avoid this for CWL with Docker, but failed for CWL with Singularity, because `cwl --singularity` does not allow to configure the `--home` option (which is good to ensure the portability, but not suitable for this kind of bad behaving tool). I will ask the developers of sra-tools to fix the problem but assume it takes time (or never happens).

So, in short, you cannot run this workflow with the combination of cwltool and Singularity.

Instead, here's a workaround: you can download the `.sra` file with the CWL tool [`download-sra`](https://github.com/pitagora-network/pitagora-cwl/blob/master/tools/download-sra/download-sra.cwl), and run sra-tools via Singularity by hand. It'd be like this:

```
$ singularity pull docker://ncbi/sra-tools:2.11.0
$ singularity exec sra-tools_2.11.0.sif vdb-config --interactive
# You can just save and exit, the config file will be saved at $HOME/.ncbi/user-settings.mkfg
$ singularity exec sra-tools_2.11.0.sif fasterq-dump SRR1274306.sra --threads 8 --skip-technical --split-files --split-spot
```

I'm sorry for the inconvenience. Hope this helps.

**Comment by s-tahara** [comment]
@inutano Thank you so much for your helpful and kind reply!

I understood the complex problem.
I appreciate your kindness and effort.

Anyway, I could get fastq files as you recommended.
Thank you so much!!

**Comment by inutano** [comment]
I've been trying to fix this but couldn't solve the compatibility issue of CWL/Singularity/SRAtools. As far as I've tested, the combinations that work are below:

- cwltool/Docker/SRAtools = OK, but need `--no-match-user`
- Singularity/SRAtools = OK, but need to configure first as above
- cwltool/Singularity/SRAtools = NG, cwltool needs to overwrite `HOME`, but Singularity won't allow, and SRAtools will use that `HOME`

Very sorry for the inconvenience, will let you know if I could find a solution for this.
